### ISSUE TITLE
Poor performance of `BitVec` initialization

### ISSUE THREAD

**Issue by caelunshun** [myrrlyn/bitvec#28]
Thanks for this library!

I have noticed an interesting performance issue with the initialization of `BitVec`. The following benchmarks produce an appalling result:

```rust
extern crate test;
use test::Bencher;
use bitvec::vec::BitVec;

#[bench]
fn bench_bitvec_init(b: &mut Bencher) {
    b.iter(|| bitvec![0; 16 * 16 * 9]);
}

#[bench]
fn bench_vec_init(b: &mut Bencher) {
    b.iter(|| vec![0u8; 16 * 16 * 9 / 8]);
}
```

Results:
```
running 2 tests
test tests::bench_vec_init       ... bench:          47 ns/iter (+/- 11)
test tests::bench_bitvec_init ... bench:      63,721 ns/iter (+/- 3,576)
```

**Comment by myrrlyn** [comment]
Thanks for bringing this to my attention; I hadn't touched the macros in a long time and there's definitely a better implementation available for `bit; count` since then. 

Until I get it landed, `BitVec::from_vec` or `[0; N].bits().to_owned()` ought to be faster for slab creation.

**Comment by myrrlyn** [comment]
Branch [hotfix/28](https://github.com/myrrlyn/bitvec/tree/hotfix/28) fixes this, as does [crate release 0.15.2](https://crates.io/crates/bitvec/0.15.2). Thanks for the report!
